src/amount.cpp

```diff
@@ -19,10 +19,10 @@ CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nSize)
 
 CAmount CFeeRate::GetFee(size_t nSize) const
 {
-    CAmount nFee = nSatoshisPerK*nSize / 1000;
+    CAmount nFee = nSatoshisPerK * nSize / 1000;
 
-    if (nFee == 0 && nSatoshisPerK > 0)
-        nFee = nSatoshisPerK;
+    if (nFee == 0 && nSize != 0 && nSatoshisPerK != 0)
```

Comment:
TBH, I'm not sure exactly what happens.  Although I thought nSatoshiPerK was a CAmount which is an int64_t.  I'm not sure it's meant to support negative rates, but at some point we need to either make sure they are supported or prevent against them.  All I'm suggesting is that right now we don't change any edge case behavior around negative amounts, b/c its already fragile.